coverage_assessment.R

```diff
@@ -41,9 +41,131 @@ nut <- get_data()
 
 # Problem 3: Prevalence of acute undernutrition/malnutrition ----
 
+####MAM
 
+with(nut, sum(muac >= 11.5 & muac <12.5  & age >=6 & age <60 & oedema == 2, na.rm= TRUE))
 
+##number of moderate acute malnourished (MAM) cases children 6-59 months old 
 
+Male_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & sex == 1 , na.rm= TRUE ))
+
+##assigning moderate acute malnourished (MAM) male 6-59 months old to Male_MAM
+
+Female_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & sex == 2 , na.rm= TRUE ))
+
+##assigning  moderate acute malnourished (MAM) female 6-59 months old to Female_MAM
+
+UM_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & county == "Urban Montserrado" , na.rm= TRUE))
+
+##assigning moderate acute malnourished (MAM) cases children 6-59 months old  in Urban Montserrado to UM_MAM
+
+GB_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 &age >=6 & age <60 & oedema == 2 & county == "Grand Bassa" , na.rm= TRUE))
+
+##assigning  moderate acute malnourished (MAM) cases children 6-59 months old other than Urban Montserrado to GB_MAM
+
+MAM <- data.frame(Male_MAM, Female_MAM, UM_MAM, GB_MAM)
+
+## creating a data frame for MAM
+
+if(interactive()) View(MAM)
+
+##Viewing MAM data
+
+
+
+####Alternate table method for number of moderate,severe and normal malnutrition by sex and by location:
+
+num_data <- with(nut, data.frame(sex,county,muac,oedema))
+
+##assigning the data frame to num_data
+
+SEX <- ifelse(num_data$sex == "1", "Male", "Female")
+OEDEMA <- ifelse(num_data$oedema == "1", "Yes", "No")
+MUAC <- with( num_data, ifelse (muac <11.5, "Severe",
```

Comment:
Two comments in this line of code:

* MUAC object name is not ideal here because the values are not MUAC values. Better call it in such a way that describes what the recoding is about. Here you are classifying each child whether they are SAM, MAM or not SAM or MAM. See how I would have done this classification below
* This is an overly complicated `ifelse()` logic with lots of nested `ifelse()` and this is usually where the misclassification happens. In your logic here, if there are children with oedema but with MUAC between 11.5 and 12.5 in the dataset, they will get classified as moderate. Also, this logic doesn't take into account the age of the children so those who are not within the age group (6-59 months), they will get classified as well.